opennlp-tools/src/main/java/opennlp/tools/chunker/ChunkerME.java

```diff
@@ -115,28 +116,50 @@ private ChunkerME(ChunkerModel model, int beamSize) {
   public ChunkerME(ChunkerModel model) {
     this(model, DEFAULT_BEAM_SIZE);
   }
-
+/////
   public String[] chunk(String[] toks, String[] tags) {
-    bestSequence = model.bestSequence(toks, new Object[] {tags}, contextGenerator, sequenceValidator);
-    List<String> c = bestSequence.getOutcomes();
-    return c.toArray(new String[c.size()]);
+    TokenTag[] tuples = TokenTag.create(toks, tags);
+    return chunk(tuples);
   }
 
   public Span[] chunkAsSpans(String[] toks, String[] tags) {
-    String[] preds = chunk(toks, tags);
-    return ChunkSample.phrasesAsSpanList(toks, tags, preds);
+    TokenTag[] tuples = TokenTag.create(toks, tags);
+    return chunkAsSpans(tuples);
   }
 
   public Sequence[] topKSequences(String[] sentence, String[] tags) {
-    return model.bestSequences(DEFAULT_BEAM_SIZE, sentence,
-        new Object[] { tags }, contextGenerator, sequenceValidator);
+    TokenTag[] tuples = TokenTag.create(sentence, tags);
+    return topKSequences(tuples);
   }
 
   public Sequence[] topKSequences(String[] sentence, String[] tags, double minSequenceScore) {
-    return model.bestSequences(DEFAULT_BEAM_SIZE, sentence, new Object[] { tags }, minSequenceScore,
-        contextGenerator, sequenceValidator);
+    TokenTag[] tuples = TokenTag.create(sentence, tags);
+    return topKSequences(tuples, minSequenceScore);
+  }
+  /////
+  public String[] chunk(TokenTag[] toks) {
```

Comment:
I don't think it makes sense to add this set of methods which take the TokenTag as input, lets keep this internal.